snapcraft_parser/main.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python3
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+snapcraft-parser
+
+Usage:
+  snapcraft-parser [options]
+
+Options:
+  -h --help                             show this help message and exit
+  -p --print                            print the master parts list
+  -v --version                          show program version and exit
+  -d --debug                            print debug information while executing
+                                        (including backtraces)
+
+
+For more help, visit the documentation:
+http://developer.ubuntu.com/snappy/snapcraft
+"""
+
+import gzip
+import logging
+import os
+import pkg_resources
+import sys
+import textwrap
+import yaml
+
+from docopt import docopt
+
+import snapcraft_parser
+from snapcraft.internal import log, sources
+
+
+logger = logging.getLogger(__name__)
+
+
+PART_NAMESPACE_SEP = '/'
+# TODO: make this a temporary directory that get's removed when finished
+BASE_DIR = "/tmp"
+PARTS_FILE = os.path.join(BASE_DIR, "snap-parts.yaml.gz")
+
+
+def _get_version():
+    try:
+        return pkg_resources.require('snapcraft-parser')[0].version
+    except pkg_resources.DistributionNotFound:
+        return 'devel'
+
+
+def main(argv=None):
+    args = docopt(__doc__, version=_get_version(), argv=argv)
+
+    # Default log level is INFO unless --debug is specified
+    log_level = logging.INFO
+    if args['--debug']:
+        log_level = logging.DEBUG
+
+    log.configure(log_level=log_level)
+
+    try:
+        return run(args)
+    except Exception as e:
+        if args['--debug']:
+            raise
+
+        sys.exit(textwrap.fill(str(e)))
+
+
+def _update_after_parts(partname, after_parts):
+    return [_namespaced_partname(partname, x) for x in after_parts]
+
+
+def _get_output():
+    # TODO: pull the part origin list from the wiki
```

Comment:
maybe make this a command line switch (what input to give it) so we can pass it hand crafted files (to test easily) or actual urls (to test or the official one)